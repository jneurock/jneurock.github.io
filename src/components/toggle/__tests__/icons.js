import React from 'react';
import ToggleIcons from '../icons';
import renderer from 'react-test-renderer';
import { afterEach, describe, expect, it } from '@jest/globals';
import { cleanup, fireEvent, render } from '@testing-library/react';

const MOCK_ICONS = [
  { component: 'div', icon: 'foo', title: 'foo' },
  { component: 'div', icon: 'bar', title: 'bar' },
];

describe('component | toggle | icons', function() {
  afterEach(function() {
    cleanup();
  });

  it('renders with selected position', function() {
    const tree = renderer
      .create(<ToggleIcons icons={MOCK_ICONS} selectedPosition={0} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('sends the selected position on click', function() {
    const handler = (_, i) => expect(i).toBe(0);

    const { container, getByTitle } = render(
      <ToggleIcons clickHandler={handler} icons={MOCK_ICONS} />
    );

    fireEvent.click(getByTitle('foo'));

    expect(container).toMatchSnapshot();
  });

  it('sends the selected position on key event', function() {
    const handler = (_, i) => expect(i).toBe(1);

    const { container, getByTitle } = render(
      <ToggleIcons keyHandler={handler} icons={MOCK_ICONS} />
    );

    fireEvent.keyDown(getByTitle('bar'), { key: 'Enter' });

    expect(container).toMatchSnapshot();
  });
});
