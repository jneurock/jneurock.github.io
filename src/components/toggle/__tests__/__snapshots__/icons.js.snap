// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | toggle-icons renders with selected position 1`] = `
<div
  className="toggle-icons flex flex--space-between"
>
  <div
    className="toggle-icon toggle-icon--active"
    icon="foo"
    onClick={[Function]}
    onKeyDown={[Function]}
    title="foo"
  />
  <div
    className="toggle-icon "
    icon="bar"
    onClick={[Function]}
    onKeyDown={[Function]}
    title="bar"
  />
</div>
`;

exports[`component | toggle-icons sends the selected position on click 1`] = `
<div>
  <div
    class="toggle-icons flex flex--space-between"
  >
    <div
      class="toggle-icon "
      icon="foo"
      title="foo"
    />
    <div
      class="toggle-icon "
      icon="bar"
      title="bar"
    />
  </div>
</div>
`;

exports[`component | toggle-icons sends the selected position on key event 1`] = `
<div>
  <div
    class="toggle-icons flex flex--space-between"
  >
    <div
      class="toggle-icon "
      icon="foo"
      title="foo"
    />
    <div
      class="toggle-icon "
      icon="bar"
      title="bar"
    />
  </div>
</div>
`;
